src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs

```diff
@@ -87,16 +88,46 @@ protected override bool IsEndOfLine(SyntaxTrivia trivia)
         protected override bool IsEndOfFileToken(SyntaxToken token)
             => token.Kind() == SyntaxKind.EndOfFileToken;
 
-        protected override SyntaxNode AddGlobalSuppressMessageAttribute(SyntaxNode newRoot, ISymbol targetSymbol, Diagnostic diagnostic, Workspace workspace, CancellationToken cancellationToken)
+        protected override SyntaxNode AddGlobalSuppressMessageAttribute(
+            SyntaxNode newRoot,
+            ISymbol targetSymbol,
+            Diagnostic diagnostic,
+            Workspace workspace,
+            Compilation compilation,
+            IAddImportsService addImportsService,
+            CancellationToken cancellationToken)
         {
             var compilationRoot = (CompilationUnitSyntax)newRoot;
             var isFirst = !compilationRoot.AttributeLists.Any();
-            var leadingTriviaForAttributeList = isFirst && !compilationRoot.HasLeadingTrivia ?
-                SyntaxFactory.TriviaList(SyntaxFactory.Comment(GlobalSuppressionsFileHeaderComment)) :
-                default;
-            var attributeList = CreateAttributeList(targetSymbol, diagnostic, isAssemblyAttribute: true, leadingTrivia: leadingTriviaForAttributeList, needsLeadingEndOfLine: !isFirst);
-            attributeList = (AttributeListSyntax)Formatter.Format(attributeList, workspace, cancellationToken: cancellationToken);
-            return compilationRoot.AddAttributeLists(attributeList);
+
+            var fullAttributeName = (QualifiedNameSyntax)SyntaxFactory.ParseName(SuppressMessageAttributeName);
+            var usingDirective = SyntaxFactory.UsingDirective(fullAttributeName.Left);
+
+            var hasUsingDirective = addImportsService.HasExistingImport(compilation, compilationRoot, contextLocation: null, usingDirective);
+            if (!hasUsingDirective && isFirst)
+            {
+                compilationRoot = Format(compilationRoot.AddUsings(usingDirective));
+                hasUsingDirective = true;
+            }
```

Comment:
❓ Can we not just use `Simplifier.AddImportsAnnotation` on the attribute?

Reply:
Agreed. Explicitly invoking the formatter here seems strange.

Reply:
@mavasani In 2015 you disabled formatting and simplifying in the fix-all and started calling the formatted here. See https://github.com/dotnet/roslyn/pull/6232/files#diff-6d2fa3555fa9d81aa6f841a1c444ab2dR49

Reply:
@sharwell I tried that first and when it didn't work I discovered the perf note linked above.

Reply:
@jnm2 Oh thanks for finding that out :-). I now recall I originally did not have the Format invocation, but invoking a FixAll caused massive performance overhead and had to fallback to invoking Format explicitly.

Reply:
Fix is here: https://github.com/dotnet/roslyn/pull/41226

Reply:
@CyrusNajmabadi That link goes to this PR lol

Reply:
Oops.  https://github.com/dotnet/roslyn/pull/41239